test/e2e/e2e_test.go

```diff
@@ -697,6 +885,108 @@ var _ = Describe("e2e external gateway validation", func() {
+			checkHybridGWPodRoutes(f, ciWorkerNodeSrc, "hybrid-ext-gw-adds-routes-e2e", true))
+	})
+
+	It("Should not add default routes over .3 if both hybrid external gw and routing-external-gws annotations are set", func() {
+		// non-ha ci mode runs a set of kind nodes prefixed with ovn-worker
+		ciWorkerNodeSrc := ovnWorkerNode
+		if haMode {
+			// ha ci mode runs a named set of nodes with a prefix of ovn-control-plane
+			ciWorkerNodeSrc = ovnHaWorkerNode
+		}
+		localVtepIP, err := runCommand("docker", "inspect", "-f", ciNetworkFlag, ciWorkerNodeSrc)
+		if err != nil {
+			framework.Failf("failed to get the node ip address from node %s %v", ciWorkerNodeSrc, err)
+		}
+		localVtepIP = strings.TrimSuffix(localVtepIP, "\n")
+		if ip := net.ParseIP(localVtepIP); ip == nil {
+			framework.Failf("Unable to retrieve a valid address from container %s with inspect output of %s", gwContainerName, localVtepIP)
+		}
+		framework.Logf("the pod side vtep node is %s and the ip %s", ciWorkerNodeSrc, localVtepIP)
+		podCIDR, err := getNodePodCIDR(ciWorkerNodeSrc)
+		if err != nil {
+			framework.Failf("Error retrieving the pod cidr from %s %v", ciWorkerNodeSrc, err)
+		}
+		framework.Logf("the pod cidr for node %s is %s", ciWorkerNodeSrc, podCIDR)
+		// setup the container to act as an external gateway and vtep
+		_, err = runCommand("docker", "exec", gwContainerName, "ip", "link", "add", "vxlan0", "type", "vxlan", "dev",
+			"eth0", "id", "4097", "dstport", vxlanPort, "remote", localVtepIP)
+		if err != nil {
+			framework.Failf("failed to create the vxlan interface on the test container: %v", err)
+		}
+		_, err = runCommand("docker", "exec", gwContainerName, "ip", "link", "set", "vxlan0", "up")
+		if err != nil {
+			framework.Failf("failed to enable the vxlan interface on the test container: %v", err)
+		}
+		_, err = runCommand("docker", "exec", gwContainerName, "ip", "address", "add", extGWCidr, "dev", "lo")
+		if err != nil {
+			framework.Failf("failed to add the external gateway ip to dev lo on the test container: %v", err)
+		}
+		_, err = runCommand("docker", "exec", gwContainerName, "ip", "route", "add", podCIDR, "dev", "vxlan0")
+		if err != nil {
+			framework.Failf("failed to add the pod route on the test container: %v", err)
+		}
+		// give the container time to come up and stabilize
+		time.Sleep(time.Second * 10)
+		annotateArgs := []string{
+			"annotate",
+			"namespace",
+			f.Namespace.Name,
+			fmt.Sprintf("k8s.ovn.org/routing-external-gws=%s", "172.17.250.1"),
+		}
+		framework.Logf("Annotating the external gateway test namespace with routing-external-gws annotation:%s", "172.17.250.1")
+		framework.RunKubectlOrDie(annotateArgs...)
+		By(fmt.Sprintf("Creating a container on %s and check default routes", ciWorkerNodeSrc))
+		framework.ExpectError(
+			// check pod has default routes going thru .3 address
+			checkHybridGWPodRoutes(f, ciWorkerNodeSrc, "hybrid-external-routing-external-gws-preference-e2e", true))
```

Comment:
I think you should just change that function to be getPodGWRoute and then compare it here in the test case to see if its .1

Reply:
Oh? I wish I had saw that function earlier...

Reply:
Ah, you mean changing the semantics of the function, I thought getPodGWRoute was an already existing function that I missed, phew.

Reply:
oh sorry, I meant refactor the checkHybridGWPodRoutes

Reply:
done